packages/fxa-payments-server/src/lib/validator.ts

```diff
@@ -1,14 +1,31 @@
 import { useReducer, useMemo } from 'react';
 
-export const useValidatorState = (): Validator => {
-  // TODO: Accept a reducer parameter to wrap actionReducer and enable overall form-level validation?
-  const [ state, dispatch ] = useReducer(mainReducer, initialState);
+export const useValidatorState = (params?: UseValidatorStateParams): Validator => {
```

Comment:
Around here is the main change to the validator state code. Adds params to accept initialState and a middleware reducer.